### PR TITLE
more pythonic code

### PR DESCRIPTION
**Pull request by milos-korenciak**
More pythonic equivalent without unnecessary local variable; this uses generator (surely c-optimized for-loop) and sum (c optimized?)

### CODE REVIEW

**Comment by milos-korenciak** [comment]
Fixing code to be more pythonic.
Maybe it can be divided into more lines, but it seems to me still perfectly readable. (Maybe I am wrong.)

File: netCDF4/utils.py [hunk 1]
```diff
@@ -190,12 +190,10 @@ def _StartCountStride(elem, shape, dimensions=None, grp=None, datashape=None,\
...
         elem = [elem]
     
-    hasEllipsis=False
-    for e in elem:
-        if type(e)==type(Ellipsis):
-            if hasEllipsis:
-                raise IndexError("At most one ellipsis allowed in a slicing expression")
-            hasEllipsis=True
+    # ensure there is at most 1 ellipse
+    if sum((1 for e in elem if type(e) == type(Ellipsis))) > 1:
```
Comment (shoyer) [inline netCDF4/utils.py]: You can omit the extra `()` inside sum.

Also, there is only one `Ellipsis` object (it's a singleton), so `e is Ellipsis` would suffice.

**Comment by jswhit** [comment]
Maybe it's just me, but I find the original code more readable (but definitely more verbose).

**Comment by willirath** [comment]
> Maybe it's just me, but I find the original code more readable (but definitely more verbose).

I agree.  What about:
```python
if elem.count(Ellipsis) > 1:
    raise IndexError("At most one ellipsis allowed in a slicing expression")
```

File: netCDF4/utils.py (hunk 1 above)
Comment (dopplershift) [reply netCDF4/utils.py]: You also don't need the extra inner set of parenthesis.

**Comment by milos-korenciak** [comment]
I definitely love willirath solution :+1:

**Comment by milos-korenciak** [comment]
So we canot use elem.count as it uses == for comparison. I.e. when used fancy indexing (= numpy array used as indexes) there occurs internal comparison "numpy.array == Ellipsis", which gives:
"ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()"
So, back to shoyer's solution.

**Comment by jswhit** [comment]
looks good, merging now - thanks!

**Pull request merged**